src/Avalonia.Base/Rendering/Composition/Server/ServerCompositionTarget.cs

```diff
@@ -62,7 +80,33 @@ internal partial class ServerCompositionTarget : IDisposable
                     v.Deactivate();
             }
         }
-        
+
+        partial void OnDebugOverlaysChanged()
+        {
+            if ((DebugOverlays & RendererDebugOverlays.Fps) == 0)
+            {
+                _fpsCounter?.Reset();
+            }
+
+            if ((DebugOverlays & RendererDebugOverlays.LayoutTimeGraph) == 0)
+            {
+                _layoutTimeGraph?.Reset();
+            }
+
+            if ((DebugOverlays & RendererDebugOverlays.RenderTimeGraph) == 0)
+            {
+                _renderTimeGraph?.Reset();
+            }
+        }
+
+        partial void OnLastLayoutPassTimingChanged()
```

Comment:
What happens if there was no layout pass for the frame? e. g. only `RenderTransform` got changed or the frame render was triggered by a render-thread animation?

Reply:
Nothing happens in this case. The `LayoutManager` isn't triggered ⇒ `LastLayoutPassTiming` doesn't change ⇒ no value is added to the graph ⇒ the layout graph doesn't change.

Reply:
So the layout graph won't show that there was zero layout time for the frame?

Also, technically more than one composition batch could have been applied before a frame rendering, how is that handled?

Reply:
The two graphs are completely independent at the moment: the layout graph only advances when there's a layout pass, same thing for the render graph when there's a render pass.

Multiple composition batches before a render should be fine: each time the batch is deserialized with a new `LastLayoutPassTiming`, that new value is added to the layout graph in `OnLastLayoutPassTimingChanged`. Later the renderer can use all the stored values to draw the layout graph with several new points in it.

In the future I'd like to add a real "total frame" graph with layout+render, including GPU if possible. In this case, there will be a need to correlate N layouts passes with a single rendered frame, with potentially large delays between the two passes.

My main motivation here was the layout graph. Since the rendering doesn't include the GPU time and the two graphs don't advance at the same time, I can remove the render graph if that makes more sense. (I personally still find the render graph useful for scenarios with custom drawing: have a look at RenderDemo's WriteableBitmap with the graph on).

Reply:
Ah, OK, so graph correlation wasn't a goal yet.

Reply:
Sorry I should have made this clear from the start.

With the two threads being independent, having two independent graphs made sense to me (especially in render-only animation demos), but I acknowledge it can also be confusing.